verification/src/compiler/compilers.rs

```diff
@@ -62,16 +63,17 @@ impl<T: Fetcher> Compilers<T> {
     }
 }
 
-impl<T: VersionList> VersionList for Compilers<T> {
-    fn all_versions(&self) -> Vec<&CompilerVersion> {
-        self.fetcher.all_versions()
+#[async_trait]
+impl<T: VersionList + Send + Sync> VersionList for Compilers<T> {
+    async fn all_versions(&self) -> Vec<CompilerVersion> {
```

Comment:
if you replace the lock to `parking_lot`, you can replace back to the usual function